src/formatDate.js

```diff
@@ -50,7 +50,63 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let DDfrom = 0;
+  let MMfrom = 0;
+  let DDto = 0;
+  let MMto = 0;
+  const sep = toFormat[3];
+  const interimObject = {};
+  const lastArray = [];
+  const dateArray = date.split(fromFormat[3]);
+
+  for (let iFrom = 0; iFrom < fromFormat.length - 1; iFrom++) {
+    for (let iTo = 0; iTo < toFormat.length - 1; iTo++) {
+      if (fromFormat[iFrom] === toFormat[iTo] && fromFormat[iFrom] === 'DD') {
+        DDfrom = iFrom;
+        DDto = iTo;
+      }
```

Comment:
It's workable solution but too complex.
You don't need to use any cycles. Instead of them you can simple use `.indexOf()` method for fromFormatArray and toFormatArray.